src/Couchbase.Lite.Shared/API/Database/Scope.cs

```diff
@@ -1,40 +1,184 @@
+        /// </summary>
+        [NotNull]
+        internal Database Database { get; set; }
+
+        [NotNull]
+        internal ThreadSafety ThreadSafety { get; }
+
+        #endregion
+
+        #region Constructors
+
+        internal Scope([NotNull] Database database, string scope)
         {
-            throw new NotImplementedException();
+            Database = database;
+            ThreadSafety = database.ThreadSafety;
+
+            Name = scope;
         }
 
-        public void RemoveChangeListener(ListenerToken token)
+        #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Gets one collection of the given name
+        /// </summary>
+        /// <param name="name">The collection name</param>
+        /// <returns>null will be returned if the collection doesn't exist in the Scope</returns>
+        public ICollection GetCollection(string name)
         {
-            throw new NotImplementedException();
+            var colls = Collections as List<ICollection>;
+            return colls.SingleOrDefault(x => x.Name == name);
         }
 
-        public IQuery CreateQuery([NotNull] string queryExpression)
+        /// <summary>
+        /// Get all collections in this scope object.
+        /// </summary>
+        /// <returns>All collections in this scope object</returns>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        public IReadOnlyList<ICollection> GetCollections()
         {
-            throw new NotImplementedException();
+            ThreadSafety.DoLocked(() =>
+            {
+                if (c4Db == null) {
+                    (Collections as IList<ICollection>).Clear();
+                    throw new InvalidOperationException(CouchbaseLiteErrorMessage.DBClosed);
+                }
+
+                var arrColl = Native.c4db_collectionNames(c4Db, Name);
+                for (uint i = 0; i < Native.FLArray_Count((FLArray*)arrColl); i++) {
+                    var collStr = (string)FLSliceExtensions.ToObject(Native.FLArray_Get((FLArray*)arrColl, i));
+                    if (GetCollection(collStr) == null) {
+                        var col = new Collection(Database, collStr, Name);
+                        col.CreateCollection();
+                        (Collections as IList<ICollection>).Add(col);
+                    }
+                }
+
+                Native.FLValue_Release((FLValue*)arrColl);
+            });
+
+            return Collections;
         }
 
-        public ICollection GetCollection(string name)
+        #endregion
+
+        #region Internal Methods
+
+        internal bool Add(ICollection collection)
         {
-            throw new NotImplementedException();
+            var res = (collection as Collection).CreateCollection();
+            if(res)
+                (Collections as IList<ICollection>).Add(collection);
```

Comment:
This is a bit clumsy looking, it might be stylistically better to have an explicit backing variable for the property that the class can use.  Or at least some AsInternal method to avoid doing this casting all over the place.